Dockerfile

```diff
@@ -1,14 +1,29 @@
-FROM alpine:3.21
-
 ARG RUBYGEM_R10K=5.0.0
 ARG RUBYGEM_PUPPET=8.10.0
 
+FROM alpine:3.21 AS builder
```

Comment:
Speaking of which, please fully qualify the Docker images; otherwise, you'll have problems building with Podman/Buildah.